packages/graph/hash_graph/lib/graph/src/ontology/types/entity_type/mod.rs

```diff
@@ -140,6 +143,40 @@ impl EntityType {
         })
     }
 
+    /// Validates the [`Entity`] against this `EntityType` schema.
+    ///
+    /// # Errors
+    ///
+    /// - [`UnknownProperty`] if the entity specifies a property, which is not defined in the entity
+    ///   type schema,
+    /// - [`MissingRequiredProperty`] if the entity type schema defines a required property which is
+    ///   not present in the entity
+    ///
+    /// [`UnknownProperty`]: ValidationError::UnknownProperty
+    /// [`MissingRequiredProperty`]: ValidationError::MissingRequiredProperty
+    pub fn validate(&self, entity: &Entity) -> Result<(), ValidationError> {
+        let allowed_properties = self.properties();
+        let properties = entity.properties();
+
+        #[allow(clippy::for_kv_map, reason = "We need to check the property as well")]
+        for (property_uri, _property) in properties {
+            ensure!(
+                allowed_properties.contains_key(property_uri),
```

Comment:
This has already made me wonder about optimisations like enforcing that keys are sorted. Anyway I agree with how we're doing it right now :)